src/main/scala/is/hail/variant/Variant.scala

```diff
@@ -366,6 +367,51 @@ case class Variant(contig: String,
     0
   }
 
+  def unifyAltAlleles(other: Variant): Tuple2[Variant, Variant] = {
+
+    if (this.locus.compare(other.locus) != 0)
+      fatal("Allele representation can only be unified for variants at the same locus.")
+
+    val (longer, shorter, swapped) = if (ref.length > other.ref.length) (this, other, false) else (other, this, true)
+    val ref_diff = longer.ref.substring(shorter.ref.length)
+
+    if (longer.ref.substring(0, shorter.ref.length) != shorter.ref)
+      fatal(s"Variants ref bases mismatch when attempting to unify alleles for variants: ${ this } and ${ other }")
+
+    val unified_shorter = shorter.copy(
+      altAlleles = shorter.altAlleles.map { aa => aa.copy(ref = longer.ref, alt = aa.alt + ref_diff) }
+    )
+
+    if (swapped)
+      (unified_shorter, longer)
+    else
+      (longer, unified_shorter)
+  }
+
+  def alleleMapWithVariant(other: Variant) : IndexedSeq[Option[Int]] = {
+
+    val (thisUnified, otherUnified) = unifyAltAlleles(other)
+    val oldAlleleIndexMap = otherUnified.altAlleles.map(_.alt).zipWithIndex.toMap
+
+    thisUnified.altAlleles.map( aa => oldAlleleIndexMap.get(aa.alt))
+
+  }
+
+  def altAlleleIndex(aa: AltAllele): Int = {
```

Comment:
definitely on static object - this doesn't do what I would expect from the name (`this.altAlleles.index(aa)`)